tests/test_forms.py

```diff
@@ -725,6 +725,67 @@ def validate_some_slot(self, value, dispatcher, tracker, domain):
     assert events == [SlotSet("some_slot", "validated_value")]
 
 
+def test_validate_prefilled_slots():
+    # noinspection PyAbstractClass
+    class CustomFormAction(FormAction):
+        def name(self):
+            return "some_form"
+
+        @staticmethod
+        def required_slots(_tracker):
+            return ["some_slot", "some_other_slot"]
+
+        def validate_some_slot(self, value, dispatcher, tracker, domain):
+            if value == "some_value":
+                return {"some_slot": "validated_value"}
+            else:
+                return {"some_slot": None}
+
+    form = CustomFormAction()
+
+    tracker = Tracker(
+        "default",
+        {"some_slot": "some_value", "some_other_slot": "some_other_value"},
+        {
+            "entities": [{"entity": "some_slot", "value": "some_bad_value"}],
+            "text": "some text",
+        },
+        [],
+        False,
+        None,
+        {},
+        "action_listen",
+    )
+
+    events = form._activate_if_required(dispatcher=None, tracker=tracker, domain=None)
+    # check that the form was activated and prefilled slots were validated
+    assert events == [
+        Form("some_form"),
+        SlotSet("some_slot", "validated_value"),
+        SlotSet("some_other_slot", "some_other_value"),
+    ] or events == [
+        Form("some_form"),
+        SlotSet("some_other_slot", "some_other_value"),
+        SlotSet("some_slot", "validated_value"),
+    ]
+
+    events.extend(
+        form._validate_if_required(dispatcher=None, tracker=tracker, domain=None)
+    )
+    # check that entities picked up in input overwrite prefilled slots
+    assert events == [
+        Form("some_form"),
+        SlotSet("some_slot", "validated_value"),
+        SlotSet("some_other_slot", "some_other_value"),
+        SlotSet("some_slot", None),
+    ] or events == [
```

Comment:
why `or`? I don't think there is any stochasticity

Reply:
that's what i thought too, and then the travis test failed because some_other_slot came before some_slot